docs/dev/dev-env.md

```diff
@@ -15,47 +17,46 @@ pipx install pdm
 pipx install pre-commit
 ```
 
-## Environment Setup
+## Setup
 
-```{note}
-A universal `Makefile` is located at the root directory of the repo, and all `make` related commands are supposed to run there.
-```
-
-There is an all-in-one command to setup environment for daily development.
+Development environment can be setup with the following command:
 
 ```bash
 make dev
 ```
 
 This command will accomplish the following tasks:
 
-- Create a virtual environment if `pdm` is available. If CI context is detected, the creation will be skipped.
-- Install the project in editable mode with the dependencies for documentation, lint, package and test.
-- Install Git hooks for various kinds of check at `pre-push` stage.
+- Create a virtual environment.
+- Install all the dependencies, including those for documentation, lint, package and test.
+- Install the project in editable mode.
+- Install git hook scripts for `pre-commit`.
+
+To speed up the setup process in certain scenarios, you may find <project:#partial-setup> helpful.
 
-## Environment Cleanup
+## Cleanup
 
-In daily development, especially when we need to upgrade or add new dependencies, we tend to encounter environment-related problems. A straightforward solution is to cleanup the current environment and setup a new one with a command like `make dev`. Three different levels of cleanup approach are recommended here.
+When encountering environment-related problems. A straightforward solution is to cleanup the environment and setup a new one. Three different levels of cleanup approach are provided here.
 
 ### Intermediate cleanup
 
-This command will only remove common intermediate files, such as generated documentation, package, coverage report, cache files for mypy, pytest, ruff and so on.
+Intermediate cleanup will only remove common intermediate files, such as generated documentation, package, coverage report, cache files for mypy, pytest, ruff and so on.
 
 ```bash
 make clean
 ```
 
 ### Deep cleanup
 
-This command will remove the pre-commit hook and the virtual environment if created alongside the common intermediate files.
+Deep cleanup will remove the pre-commit hook and the virtual environment alongside the common intermediate files.
 
 ```bash
 make deepclean
 ```
 
 ### Complete cleanup
 
-With this command, the repo will be as if it has been re-cloned. It is useful if we want to start almost from scratch.
+Complete cleanup will make the repo as if it has been re-cloned. It is useful if we want to start from scratch.
```

Comment:
'make the repo'  maybe have made a mistakes with the expressing senses of remove or remake.

Reply:
Done, asked ChatGPT with minor refinements. :-)